src/oom/oomd-util.c

```diff
@@ -291,7 +291,11 @@ int oomd_cgroup_context_acquire(const char *path, OomdCGroupContext **ret) {
         ctx->preference = MANAGED_OOM_PREFERENCE_NONE;
 
         r = cg_get_path(SYSTEMD_CGROUP_CONTROLLER, path, "memory.pressure", &p);
-        if (r < 0)
+        if (r == -EOPNOTSUPP)
```

Comment:
hmm, cg_get_path() only determine sthe path, it doesn't actually read the attribute? what am i missing?

Reply:
also, better use ERRN_IS_NOT_SUPPORTED() mcro

Reply:
yep you're right, the error is from read_resource_pressure right below this, I'll fix that.

For context, this only happens on the Ubuntu CI because someone requested that PSI default to off for s390x (http://pad.lv/1876044), and currently for some reason all our s390x cloud instances are very hard to get time on...but I can repro this on plain-old intel by just using psi=0 boot param so i'll be sure to verify the change before pushing.

Thanks!

Reply:
Oh hmmm I just looked at the test again and it already has `is_pressure_supported()` which checks if /proc/pressure files exist.. So /proc/pressure files exist but cgroup2 pressure files do not? Are the files under /proc/pressure there but return nothing? If so maybe it makes sense to fix `is_pressure_supported()` to check the correct statie.

Reply:
> Oh hmmm I just looked at the test again and it already has `is_pressure_supported()` which checks if /proc/pressure files exist.. So /proc/pressure files exist but cgroup2 pressure files do not? Are the files under /proc/pressure there but return nothing? If so maybe it makes sense to fix `is_pressure_supported()` to check the correct state. Or add an additional `is_cgroup_pressure_supported()` if /proc/pressure files look sane.

For whatever reason, the kernel devs chose to leave the pressure files in place with psi=0, but any attempts to read them returns -EOPNOTSUPP. So just checking for existance of /proc/pressure/* isn't enough, at least one of them needs to be read to verify they don't return the error.

However it's probably better to just adjust is_pressure_supported() to check that instead of doing anything here? I'll change the commit to do that unless you object.

Reply:
> However it's probably better to just adjust is_pressure_supported() to check that instead of doing anything here? I'll change the commit to do that unless you object.

Yes please that would be preferred 🙂